app/controllers/opportunities_controller.rb

```diff
@@ -0,0 +1,71 @@
+class OpportunitiesController < ApplicationController
+  before_action :set_opportunity, only: [:edit, :update, :show, :destroy]
+  before_action :require_company_owner, only: [:edit, :update, :destroy]
+
+  def index
+    @opportunities = Opportunity.all
+  end
+
+  def new
+    # redirect to root path if user has no company yet
+    if current_user.companies.empty?
+      redirect_to new_company_path
+      flash[:danger] = "You must create a company before adding an internship"
+    end
+
+    @opportunity = Opportunity.new
```

Comment:
A more secure way of doing this and ensuring that the opportunities are linked to a company is something like `current_user.companies.find(params[:company_id]).oppertunities.new` . It's all object oriented but ensured that you only allow the current user to create an opportunity for the company that they have permission for and it also belongs to a company. This adds the foreign keys automatically instead of you doing this in the `create` method below.